libraries/engage/cookie-consent/subscriptions/cookieConsent.js

```diff
@@ -0,0 +1,52 @@
+import {
+  appDidStart$,
+} from '@shopgate/engage/core';
+import { finishCookiesConsent, showCookieConsentModal } from '../action-creators';
+import { getAreComfortCookiesSelected, getAreStatisticsCookiesSelected, getIsCookieConsentHandled } from '../selectors';
+import { appConfig } from '../../index';
+import { comfortCookieActivated$, statisticsCookiesActivated$ } from '../streams';
+
+/**
+ * show cookie consent modal at first app start
+ * @param {Function} subscribe The subscribe function
+ */
+export default function cookieConsent(subscribe) {
+  subscribe(appDidStart$, async ({ dispatch, getState }) => {
+    const { cookieConsent: { cookieConsentActivated } } = appConfig;
+    const isCookieConsentHandled = getIsCookieConsentHandled(getState());
+    const areStatisticsCookiesSelected = getAreStatisticsCookiesSelected(getState());
+    const areComfortCookiesSelected = getAreComfortCookiesSelected(getState());
+
+    // if merchant has not activated the cookie feature:
+    // trigger stream to continue with push opt-in modal
+    if (!cookieConsentActivated) {
+      dispatch(finishCookiesConsent({
+        areComfortCookiesSelected: null,
+        areStatisticsCookiesSelected: null,
+      }));
+    }
+
+    // if merchant has activated cookie feature but user has not chosen cookies yet:
+    // show cookie consent modal to make user choose them
+    if (cookieConsentActivated && !isCookieConsentHandled) {
+      await dispatch(showCookieConsentModal());
```

Comment:
the modal doesn't re-appear when one navigates to the details page and presses the back button there.

Reply:
fixed